### ISSUE TITLE
QR code from F-Droid doesn't open up F-Droid automatically

### ISSUE THREAD

**Issue by taivlam** [mollyim/mollyim-android#40]
* Issue

If a user has the Molly repo enabled in F-Droid and scans the QR code with another device, the second device is lead to an intermediary Molly webpage, instead of opening up F-Droid with the repo details filled in and waiting for confirmation to be added to F-Droid.

* Details

I have 1 device with the Molly-FOSS repository enabled, and I have a second device that is new.  I want to install Molly-FOSS on the second device.  When the second device scans the QR code for the Molly repo on the first device, the second device opens up the browser to [this page](https://molly.im/fdroid/foss/repo/?fingerprint=5198DAEF37FC23C14D5EE32305B2AF45787BD7DF2034DE33AD302BDB3446DF74) instead of opening F-Droid with the custom repo information filled out.  I end up going to the [official Molly page](https://molly.im/fdroid/) and scanning the corresponding QR code for Molly-FOSS.

The former QR code from inside F-Droid encodes the intermediary page,
`https://molly.im/fdroid/foss/repo?fingerprint=5198DAEF37FC23C14D5EE32305B2AF45787BD7DF2034DE33AD302BDB3446DF74`;
while the latter QR code encodes an additional prefix to open F-Droid with:
`fdroidrepos://molly.im/fdroid/foss/repo?fingerprint=5198DAEF37FC23C14D5EE32305B2AF45787BD7DF2034DE33AD302BDB3446DF74`.

* Expected Behavior

It was expected that the second device would automatically open up F-Droid with the Molly-FOSS details filled in and ready to the user to either add repo or cancel this action.

* Other Notes

I was transferring from an old device to a new device (both unofficial LOS 18.1 for microG).  I was using Binary Eye as my QR scanner and Bromite was my browser.

I thought the behavior of the Molly QR code (from inside F-Droid) was not as expected, because I was able to automatically open up F-Droid with repo information prefilled when I scanned the corresponding QR codes for the following apps with custom F-Droid repos: Bitwarden, Briar, Bromite, Haven (from the Guardian Project), and Newpipe.

(This is coming from a perspective of convenience - if I was transferring my apps to a new device, then it would be the quickest to simply install the app on the new device by scanning the repo QR code from F-Droid on the old device, instead of opening up the webpage again on my first device or my laptop.  Molly-FOSS was the only app that I ended up needing to scan the from the webpage.)

I'm not sure if this is the right place to submit this, since it's not about the Molly app itself.

Also I am not sure if this behavior was intentionally designed or if the QR code shouldn't be doing this.

**Comment by taivlam** [comment]
I forgot to mention this: the 5 other apps with custom F-Droid repos also had "ordinary looking" URLs (so those QR codes were like the former QR codes),

**Comment by valldrac** [comment]
> I'm not sure if this is the right place to submit this, since it's not about the Molly app itself.

You did well. It could be a mistake in the metadata information of Molly's repository [here](https://molly.im/fdroid/foss/repo/index.xml). I'll check it out.

Thanks for the report.

**Comment by valldrac** [comment]
I found the problem. Our FOSS repository is not compatible with f-droid. The URL doesn't match any of the supported paths https://gitlab.com/fdroid/fdroidclient/-/blob/master/app/src/main/AndroidManifest.xml

Hard to fix it now, because we have to keep the broken URL working for old clients.

This issue will be blocked until we replace GitHub pages by a proper hosting to store the repository.

**Comment by valldrac** [comment]
> This issue will be blocked until we replace GitHub pages by a proper hosting to store the repository.

Never mind. I think I found a workaround to fix it.

**Comment by taivlam** [comment]
Thank you!

I just tested this and the QR code in the F-Droid repo now works properly.

Well, there is a side note: if you still have the older Molly (FOSS) repo in F-Droid before this change was pushed, then the F-Droid repo still behaves as described above.  To see the change, I had to remove the older F-Droid repo and re-add it from the [Molly repo site](https://molly.im/fdroid/) (since the current QR code has visually changed from the previous one - this should be obvious after some thinking, but I just wanted to take note of this in the case someone still has the prior F-Droid repo entry on their respective device.)
